### ISSUE TITLE
Multiple Incorrect warnings for Argument type is any

### ISSUE THREAD

**Issue by vikigenius** [DetachHead/basedpyright#1119]
### Description

Simple to reproduce

```python
from typing import Literal
from pydantic import BaseModel, Field

MyType = Literal['LiteralA', 'LiteralB', 'LiteralC']


class Model(BaseModel):
    myvals: list[MyType] = Field(default_factory=lambda: ['LiteralA'])
```

This gives me 33 warnings for Argument type is Any. Pyright gives none. This behavior seems to have been introduced very recently.

This is on 1.28.1 and python 3.12

**Comment by DetachHead** [comment]
this was caused by an upstream change in 1.1.395 where default arguments are now always validated against their type annotation (https://github.com/microsoft/pyright/pull/9922). this broke overload matching in many cases where `...` is used as a default value (https://github.com/microsoft/pyright/issues/9984), so it was reverted in 1.1.396

i actually caught this bug when merging 1.1.395 so i pushed my own fix that addressed both the original issue and the issue caused by the fix, so i dropped the hot fix from 1.1.396.

but it looks like the original change caused another problem that nobody noticed until now, where `reportAny` and `reportUnknown*` errors are reported on calls to functions where the default values are `Any`/`Unknown even if they aren't used in the call. here's a minimal repro:

```py
def foo(_asdf: int = asdf): ... #pyright:ignore[reportUndefinedVariable]

_ = foo() # error: Argument type is unknown (reportUnknownArgumentType)
```

this is happening in your code because pydantic uses this `_Unset` value for the default values to `Field`:

```py
_Unset: Any = PydanticUndefined
```

i'll see if there's an easy way to fix this while still addressing the original issue. if not, i'll just revert the whole change like eric did in 07668effd5901360c330e30d680bc1e3cad3b161
